packages/ui/Cancelable/Cancelable.js

```diff
@@ -216,6 +215,11 @@ const Cancelable = hoc(defaultConfig, (config, Wrapped) => {
 			delete props.onCancel;
 			delete props[onCancel];
 
+			if (this.isFirstRender && modal) {
```

Comment:
This is a clever way to simulate `componentWillMount`! I think we'd like to keep our render methods [idempotent](https://en.wikipedia.org/wiki/Idempotence) but I think this is an acceptable deviation for the time being. We might also think of other ways to trigger subscription to the events rather than tied to rendering (e.g. when an event is fired similar to our discussions on #2070).

Reply:
Fun fact. `react-transition-group` has similar stuff here https://github.com/reactjs/react-transition-group/blob/master/src/TransitionGroup.js#L73